### PR TITLE
NixOS Manual: document assertions and warnings

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

assertions and warnings in NixOS modules were not documented yet.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/doc/manual/development/assertions.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+</para>
+
+<programlisting>
+<![CDATA[
+{
+  config.warnings = [
+        "Importing <nixpkgs/nixos/modules/programs/virtualbox.nix> is "
+      + "deprecated, please use `virtualisation.virtualbox.host.enable = true' "
+      + "instead." ];
+  config.virtualisation.virtualbox.host.enable = true;
```
Comment (fpletz) [inline nixos/doc/manual/development/assertions.xml]: Even though this is example is very concise, having the same structure here as below for assertions would be more clearer I think.

**Review by fpletz — APPROVED** [review]
Thanks a lot! Docs are always very welcome! Looks good to me except the comment.

**Review by domenkozar — APPROVED** [review]

**Comment by roberth** [comment]
Ok, I have simplified the assertion example and I have adapted the warning example to match the assertion example more closely. In doing so, the warning example is not a real world example any more. It turns out that they are not used very often. Most of the warnings seem to be deprecations, moves and such, which are modeled differently.

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**